mapq0_vcf_filter.sh

```diff
@@ -4,49 +4,26 @@ set -e
 outvcf=$1
 vcf=$2
 bam=$3
-ref_fasta=$4
-mapq0perc=$5
+mapq0perc=$4
+sample_name=$5
 outdir=$(dirname "$outvcf")
 
-#grab sites that don't already have the MQ0 field
-zgrep -v "^#" "$vcf" | grep -v "MQ0" | cut -f 1,2 | while read chr pos;do
+count=0;
+zgrep -v "^#" "$vcf" | cut -f 1,2 | while read chr pos;do
     pysamstats --type mapq --chromosome $chr --start $pos --end $((pos+1)) "$bam"  | grep $pos | cut -f 1,2,5 >>$outdir/mapq0counts
+    count=$((count+1))
 done
 
 
-if [[ ! -s $outdir/mapq0counts ]];then
-    #no sites to process, just make a copy of the vcf. 
-    #Have to handle both gzipped and unzipped vcfs
+if [[ $count -eq 0 ]];then
+    #no sites to process, just make a copy of the (empty) vcf.
+    #handle both gzipped and unzipped vcfs
     if [[ ${vcf: -3} == ".gz" ]];then 
-        gunzip -c $vcf > $outdir/mapq0.vcf
+        gunzip -c $vcf > $outdir/mapq_filtered.vcf.gz
     else
-        cp $vcf $outdir/mapq0.vcf
+        cp $vcf $outdir/mapq_filtered.vcf.gz
```

Comment:
I don't think it make sense to run `gunzip` and then name the file as `.vcf.gz` or to unilaterally add `.gz` if it didn't have it before.

Reply:
Thanks - got this logic flipped